opm/autodiff/WellStateFullyImplicitBlackoilSolvent.hpp

```diff
@@ -33,6 +33,37 @@ namespace Opm
         /// One solvent fraction per well connection
         std::vector<double>& solventFraction() { return solvent_fraction_; }
         const std::vector<double>& solventFraction() const { return solvent_fraction_; }
+
+        data::Wells report(const PhaseUsage &pu) const override {
+            data::Wells res = WellStateFullyImplicitBlackoil::report(pu);
+
+            const int nw = WellState::numWells();
+
+            // If there are now wells numPhases throws a floating point
+            // exception.
+            if (nw == 0) {
+                return res;
+            }
+
+            const int np = BaseType::numPhases();
+
+            assert( np == 3 ); // the solvent model assumes 3 phases in the base model
+
+            // completions aren't supported yet
+            for( auto w = 0; w < nw; ++w ) {
+                using rt = data::Rates::opt;
+                double solvent_well_rate = 0.0;
+                for (int perf = wells_->well_connpos[w]; perf < wells_->well_connpos[w+1]; ++perf ) {
+                    auto solvent_rate_this = BaseType::perfPhaseRates()[np*perf + 2] * solventFraction()[perf];
```

Comment:
The `2` here makes me uneasy, I think it should use a phase position and not be hardcoded. In any case magic constants and indices should be named.